File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        print(nums)
        m = defaultdict(int)
        for i, v in enumerate(nums):
            if (target - v) in m:
                return [i, m.get(target - v)]
            else:
                m[v] = i
        return []

File: 0001-two-sum/0001-two-sum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    Map<int,int> seen = Map<int,int>();
    for(int i= 0; i < nums.length; i++){
        final value = nums[i];
        if (seen.containsKey(target - value)){
            return [seen[target-value]!,i];
        }
        seen[value] = i;
    }
    return [];
  }
}